docs - database troubles (#25591)
metabase/metabase: docs/troubleshooting-guide/my-dashboard-is-slow.md

# My dashboard is slow

First, you'll want to make sure your browser is on friendly terms with Metabase:

- Clear your browser cache and disable all extensions before refreshing the page, or

> - try loading the dashboard in a private/incognito session. 

Reviewer: ```suggestion
- Try loading the dashboard in a private/incognito session. 
```

   - For example, create new dashboards for weekly vs. monthly metrics, or new vs. returning customers.
2. Move your cards to the new dashboard(s) until each dashboard has 10 or fewer cards.
2. Use [custom destinations](https://www.metabase.com/learn/dashboards/custom-destinations) to link related cards on different dashboards.

**Explanation**

A dashboard with 50 cards is almost always going to be slower than 5 dashboards with 10 cards. Metabase displays a dashboard by refreshing all of the questions on it (that is, re-executing all of the queries against your database). Your data warehouse may try to run these queries at the same time to return the results faster, however, these queries can actually compete with each other and slow things down (like having too many programs open on your computer at once).

Aside from running faster, a small and focused dashboard is also easier for people to understand without getting overwhelmed. For more tips, check out our article on [BI dashboard best practices](https://www.metabase.com/learn/dashboards/bi-dashboard-best-practices).

## Dashboard used by many people at the same time

> 1. Optional: Use Metabase's [auditing tools](../usage-and-performance-tools/audit.md) to look at your Metabase usage stats\*.

Reviewer: ```suggestion
1. Optional: use Metabase's [auditing tools](../usage-and-performance-tools/audit.md) to look at your Metabase usage stats\*.
```

2. Set up [caching](../configuring-metabase/caching.md) for the questions on your popular dashboard.
3. Run the dashboard during off-hours so that people can load the cached results in seconds instead of executing a fresh query to your database.

\* Available on paid plans.

**Explanation**

> If you have a popular, yet slow dashboard that many people check around the same time (e.g., at the start of the work day), you may end up with queued queries or saturated database connections. Caching allows you to prepare for peak traffic by running those slow queries once, ahead of time, so that the results are ready to go.

Reviewer: ```suggestion
If you have a dashboard that many people check around the same time (e.g., at the start of the work day), you may end up with queued queries or saturated database connections. Caching allows you to prepare for peak traffic by running those slow queries once, ahead of time, so that the results are ready to go.
```

Caching takes less effort because it doesn't involve any changes to your schemas or databases. If you're ready to invest more resources into the root cause of dashboard performance issues, check out this list of common [schema and database optimizations](https://metabase.com/learn/administration/making-dashboards-faster#organize-data-to-anticipate-common-questions).

## Embedded dashboard is slow compared to original dashboard

> 1. Set up a [locked parameter to pre-filter your data](../embedding/signed-embedding-parameters.md#pre-filtering-data-in-a-signed-embed).

Reviewer: ```suggestion
1. To speed up the embedded dashboard, set up a [locked parameter to pre-filter your data](../embedding/signed-embedding-parameters.md#pre-filtering-data-in-a-signed-embed).
```

**Explanation**

One of the easiest ways to make a question or dashboard run faster is to work with a smaller dataset. Your Metabase admin can apply automatic data limitations using things like [SSO](../people-and-groups/start.md#setting-up-single-sign-on-sso), [data permissions](../permissions/data.md), and [data sandboxing](../permissions/data-sandboxes.md).

> However, a standalone embedded dashboard (a [signed embed](../embedding/signed-embedding.md)) won't log people into Metabase. This means that people viewing the signed embed won't be subject to the permissions and data restrictions set up by your admin. When someone loads a question or a dashboard in a signed embed, that question or dashboard will utilize a larger, unrestricted dataset rather than a smaller, permissions-limited dataset.

Reviewer: ```suggestion
When someone loads a question or a dashboard in a signed embed, however, that question or dashboard will query the full dataset (rather than a smaller dataset limited by permissions). Standalone, [signed embeds](../embedding/signed-embedding.md) don't require people to be logged in, and unauthenticated people viewing the signed embed won't be subject to the permissions and data restrictions set up by your admin. 
```

## Dashboard is slow compared to similar dashboards

1. Remove fields (columns) that you don't need in the final result.
2. Add a [filter](../questions/query-builder/introduction.md#filtering) to reduce the amount of data being queried. For example:
   - Narrow down the time frame to the reporting period that you care about.
   - Exclude invalid records, such as: blanks, nulls, or rows with values like "cancelled", "expired", "invalid", and so on.
3. Remove [joins](../questions/query-builder/introduction.md#joining-data) to tables that aren't being used. 
4. If you're aggregating data from the query builder, ask your database admin if there's a pre-aggregated [view](https://www.metabase.com/glossary/view) that you can use instead.

**Explanation**

> When you update your question to use a minimal number of rows or columns (or switch your question to use a smaller table, such as a summary table) your database can spend less time scanning those records in order to return your results. This is especially important to think about if you're [starting from someone else's saved question or model](../questions/query-builder/introduction.md#play-around-with-saved-questions), because you might not need all of the data that the original creator decided to include.

Reviewer: ```suggestion
When you update your question to use a minimal number of rows or columns (or switch your question to use a smaller table, such as a summary table) your database can spend less time scanning those records in order to return your results. Narrowing the scope of your question is especially important to think about if you're [starting from someone else's saved question or model](../questions/query-builder/introduction.md#play-around-with-saved-questions), because you might not need all of the data that the original creator decided to include.
```
